### PR TITLE
Engine process

### PR DESCRIPTION
**Pull request by Domonion**
Moved engine process out of the main idea process. Changes include:
1. Reactive settings
2. Removing all JDK9+ api calls to support out-of-the process execution on JDK8
3. Refactored client protocol initialization, ClientProcessBuilder introduced
4. Big refactoring

## Type of Change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: utbot-framework-api/src/main/kotlin/org/utbot/framework/UtSettings.kt [hunk 1]
```diff
@@ -282,12 +259,12 @@ object UtSettings : AbstractSettings(
...
 
     /**
-     * Determines whether should errors from a child process be written to a log file or suppressed.
+     * Determines whether should errors from a child process and idea engine process be written to a log file or suppressed.
      * Note: being enabled, this option can highly increase disk usage when using ContestEstimator.
      *
      * False by default (for saving disk space).
      */
-    var logConcreteExecutionErrors by getBooleanProperty(false)
+    var logConcreteExecutionErrors by getBooleanProperty(true)
```
Comment (sergeypospelov) [inline utbot-framework-api/src/main/kotlin/org/utbot/framework/UtSettings.kt:267]: This dramatically increases the amount of user's disk space used. Maybe divide the into two properties?

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/generator/UtTestsDialogProcessor.kt [hunk 2]
```diff
@@ -128,154 +122,166 @@ object UtTestsDialogProcessor {
...
 
-                                // set timeout for concrete execution and for generated tests
-                                UtSettings.concreteExecutionTimeoutInChildProcess = model.hangingTestsTimeout.timeoutMs
-
-                                UtSettings.useCustomJavaDocTags = model.commentStyle == JavaDocCommentStyle.CUSTOM_JAVADOC_TAGS
+                                for (srcClass in model.srcClasses) {
+                                    val (methods, className) = ReadAction.nonBlocking<Pair<List<ExecutableId>, String?>> {
+                                        val canonicalName = srcClass.canonicalName
+                                        // todo how bad is it?
+                                        val clazz = classLoader.loadClass(canonicalName).kotlin
```
Comment (sergeypospelov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/generator/UtTestsDialogProcessor.kt:173]: It won't allow running project with JDK-17 on IDEA with JDK-11.

**Comment by alisevych** [comment]
Found bug:
Error popup "Failed to save tests report". When OK is pressed SARIF-file is saved.
image

**Comment by alisevych** [comment]
Found bug:
Cancel Test generation is ignored - process continues up to the end.
image

File: utbot-framework/src/main/kotlin/org/utbot/framework/process/generated/.rdgen [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+file: C:\\wrk\\UTBotJava\\utbot-framework\\src\\main\\kotlin\\org\\utbot\\framework\\process\\generated\\EngineProcessProtocolRoot.Generated.kt=2022-09-26_10:20:14.481
+file: C:\\wrk\\UTBotJava\\utbot-framework\\src\\main\\kotlin\\org\\utbot\\framework\\process\\generated\\SettingsModel.Generated.kt=2022-09-26_10:20:14.487
+file: C:\\wrk\\UTBotJava\\utbot-framework\\src\\main\\kotlin\\org\\utbot\\framework\\process\\generated\\SettingsProtocolRoot.Generated.kt=2022-09-26_10:20:14.488
+file: C:\\wrk\\UTBotJava\\utbot-rd\\src\\main\\rdgen\\org\\utbot\\rd\\models\\ChildProcessModel.kt=2022-09-21_13:31:25.650
+file: C:\\wrk\\UTBotJava\\utbot-rd\\src\\main\\rdgen\\org\\utbot\\rd\\models\\EngineProcessModel.kt=2022-09-23_14:59:39.591
+file: C:\\wrk\\UTBotJava\\utbot-rd\\src\\main\\rdgen\\org\\utbot\\rd\\models\\SettingsModel.kt=2022-09-26_10:20:04.082
+noLineNumbersInComments=false
+outputFolders=C:\\wrk\\UTBotJava\\utbot-framework\\src\\main\\kotlin\\org\\utbot\\framework\\process\\generated
+packages=org.utbot.rd.models
+version=1.10
```
Comment (Markoutte) [inline utbot-framework/src/main/kotlin/org/utbot/framework/process/generated/.rdgen:12]: what this file for? looks like it shouldn't be here?

File: utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/process/ChildProcess.kt [hunk 4]
```diff
@@ -64,149 +54,65 @@ internal object HandlerClassesLoader : URLClassLoader(emptyArray()) {
...
-}
-
-private val messageFromMainTimeoutMillis: Long = TimeUnit.SECONDS.toMillis(120)
-private val synchronizer: Channel<State> = Channel(1)
-
 /**
  * Command-line option to disable the sandbox
  */
 const val DISABLE_SANDBOX_OPTION = "--disable-sandbox"
+private val defaultLogLevel = LogLevel.Debug
```
Comment (Markoutte) [inline utbot-instrumentation/src/main/kotlin/org/utbot/instrumentation/process/ChildProcess.kt:61]: Maybe, the higher level is required here?

**Comment by alisevych** [comment]
Found issue:
Concrete execution logs contain many debug information.
Log on test generation for Recursion class weights 1,5 GB (!)

File: utbot-framework/src/main/kotlin/org/utbot/framework/process/generated/.rdgen (hunk 3 above)
Comment (Domonion) [reply utbot-framework/src/main/kotlin/org/utbot/framework/process/generated/.rdgen:12]: This is rdgen hashfile. It helps to identify if models should be regenerated. Though it is unnecessary to be here, but it saves time generating rd models, so I'd prefer for it to stay

**Comment by Domonion** [comment]
rolling to https://github.com/UnitTestBot/UTBotJava/pull/1067

**Pull request closed without merging**